packages/core/src/utils.ts

```diff
@@ -831,43 +831,35 @@ export const getContractAddress = (
 export const getConstructorArgs = (
   constructorArgs: ParsedConfigVariables,
   abi: Array<Fragment>
-): {
-  constructorArgTypes: Array<string>
-  constructorArgValues: Array<ParsedConfigVariable>
-} => {
-  const constructorArgTypes: Array<string> = []
+): Array<ParsedConfigVariable> => {
   const constructorArgValues: Array<ParsedConfigVariable> = []
 
   const constructorFragment = abi.find(
     (fragment) => fragment.type === 'constructor'
   )
 
   if (constructorFragment === undefined) {
-    return { constructorArgTypes, constructorArgValues }
+    return constructorArgValues
   }
 
   constructorFragment.inputs.forEach((input) => {
-    constructorArgTypes.push(input.type)
     constructorArgValues.push(constructorArgs[input.name])
   })
 
-  return { constructorArgTypes, constructorArgValues }
+  return constructorArgValues
 }
 
 export const getCreationCodeWithConstructorArgs = (
   bytecode: string,
   constructorArgs: ParsedConfigVariables,
   abi: ContractArtifact['abi']
 ): string => {
-  const { constructorArgTypes, constructorArgValues } = getConstructorArgs(
-    constructorArgs,
-    abi
-  )
+  const constructorArgValues = getConstructorArgs(constructorArgs, abi)
+
+  const iface = new ethers.utils.Interface(abi)
 
   const creationCodeWithConstructorArgs = bytecode.concat(
-    remove0x(
-      utils.defaultAbiCoder.encode(constructorArgTypes, constructorArgValues)
-    )
+    remove0x(iface.encodeDeploy(constructorArgValues))
```

Comment:
I replaced `defaultAbiCoder.encode` with `iface.encodeDeploy` because `defaultAbiCoder.encode` would silently fail if the `constructorArgTypes` weren't defined correctly